kickstart/provision.erb

```diff
@@ -56,7 +56,7 @@ firewall --disable
 <% else -%>
 firewall --<%= os_major >= 6 ? 'service=' : '' %>ssh
 <% end -%>
-authconfig --useshadow --passalgo=sha256 --kickstart
+authconfig --useshadow --passalgo=<%= @host.operatingsystem.password_hash %> --kickstart
```

Comment:
Maybe you can fall back to sha256 if it's not set? That way we stay compatible with older foreman versions.